modules/swagger-codegen/src/main/resources/aspnetcore/controller.mustache

```diff
@@ -31,9 +31,10 @@ namespace {{packageName}}.Controllers
         [{{httpMethod}}]
         [Route("{{{basePathWithoutHost}}}{{{path}}}")]
         [ValidateModelState]
-        [SwaggerOperation("{{operationId}}")]{{#responses}}{{#returnType}}
-        [SwaggerResponse({{code}}, typeof({{&returnType}}), "{{message}}")]{{/returnType}}{{/responses}}
-        public virtual {{#returnType}}IActionResult{{/returnType}}{{^returnType}}void{{/returnType}} {{operationId}}({{#allParams}}{{>pathParam}}{{>queryParam}}{{>bodyParam}}{{>formParam}}{{>headerParam}}{{#hasMore}}, {{/hasMore}}{{/allParams}})
+        [SwaggerOperation("{{operationId}}")]{{#responses}}{{#dataType}}
+        [SwaggerResponse(statusCode: {{code}}, type: typeof({{&dataType}}), description: "{{message}}")]{{/dataType}}{{^dataType}}
```

Comment:
This (and everywhere using `dataType` rather than `returnType` in the responses array) would still need to be `typeof({{&returnType}})` as the desired return type and the actual data type may not be the same (e.g. returnType = ICollection, dataType = something that isn't ICollection).

Reply:
Isn't returnType the overall return type of the operation?
I'm looking for the type of a certain response..

Reply:
Hey @jimschubert: I just realized that the XML Comments do some of this anyways..
Can you confirm?

We could remove line 36 essentially?